api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1040,9 +1032,16 @@ private HttpResponse executePost(String bucketName, String objectName, Map<Strin
            InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
            InternalException {
     updateRegionCache(bucketName);
+    int len = -1;
+    if (data instanceof String) {
+      len = ((String) data).length();
+    }
+    if (data instanceof CompleteMultipartUpload) {
+      len = ((CompleteMultipartUpload) data).toString().length();
+    }
```

Comment:
Having execute{Head,Get,Post,Put,Delete} are just wrappers to execute() for convenience and readability across APIs  Spreading the logic of type checking will lead to confusion in future and its not a good idea.